### PR TITLE
Generate data file for hugo docs

### PR DESCRIPTION
**Pull request by mitchdraft**
BOT NOTES: 
resolves https://github.com/solo-io/solo-kit/issues/280

### CODE REVIEW

File: pkg/code-generator/docgen/cross_project.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			return err
+		}
+		// Collect names of files that contain resources for which doc gen has to be skipped
+		skipMap := make(map[string]bool)
+		for _, res := range project.Resources {
+			if res.SkipDocsGen {
+				skipMap[res.Filename] = true
+			}
+		}
+		protoFiles := protokit.ParseCodeGenRequest(project.Request)
```
Comment (marcogschmidt) [inline pkg/code-generator/docgen/cross_project.go]: Similar in spirit to the above nit: in `parser.parser.go:37` we already call this function with the same argument (the `req` that is used there is then set as the `Project.Request` that you use here). If we update `model.Project` to have a field like this

```go
Descriptors []*FileDescriptor
```
we could reuse the result of the previous function call and this line would become:

```go
protoFiles := project.Descriptors
```

File: pkg/code-generator/docgen/cross_project.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		return err
+	}
+	if err := ioutil.WriteFile(path, []byte(file.Content), 0644); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+// util for populating the Hugo ProtoMap to enable allows lookup by
```
Comment (marcogschmidt) [inline pkg/code-generator/docgen/cross_project.go]: Super nit: fix comment

**Comment by mitchdraft** [comment]
good idea @marcogschmidt , that cut out quite a few duplicate calls

**Comment by marcogschmidt** [comment]
```
Expected error:
Step #6 - "test":       <*errors.StatusError | 0xc000571e60>: {
Step #6 - "test":           ErrStatus: {
Step #6 - "test":               TypeMeta: {Kind: "", APIVersion: ""},
Step #6 - "test":               ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
Step #6 - "test":               Status: "Failure",
Step #6 - "test":               Message: "Operation cannot be fulfilled on configmaps \"test-lock\": the object has been modified; please apply your changes to the latest version and try again",
Step #6 - "test":               Reason: "Conflict",
Step #6 - "test":               Details: {Name: "test-lock", Group: "", Kind: "configmaps", UID: "", Causes: nil, RetryAfterSeconds: 0},
Step #6 - "test":               Code: 409,
Step #6 - "test":           },
Step #6 - "test":       }
Step #6 - "test":       Operation cannot be fulfilled on configmaps "test-lock": the object has been modified; please apply your changes to the latest version and try again
Step #6 - "test":   not to have occurred
```

/kick

**Pull request merged**